ADT75 driver + demo project

CiprianRegus opened the pull request.

amiclaus commented on drivers/temperature/adt75/adt75.c:
```diff
@@ -0,0 +1,188 @@
+ */
+static int adt75_reg_size(uint32_t addr)
+{
+	if (addr == ADT75_CONF_REG)
```
You can do this as a macro, and drop this extra function: https://github.com/torvalds/linux/blob/b7b275e60bcd5f89771e865a8239325f86d9927d/drivers/iio/dac/ad7293.c#L21-L33 .
Or just do the check inline.

amiclaus commented on include/no_os_units.h:
```diff
@@ -65,4 +65,6 @@
 #define MICROWATT_PER_MILLIWATT	1000UL
 #define MICROWATT_PER_WATT	1000000UL
 
+#define MILLIDEGREE_PER_DEGREE  1000UL
```
move to separate commit

danmois commented on projects/adt75/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,91 @@
+/***************************************************************************//**
```
There was a discussion regarding the switch from renaming the example from "DUMMY" to "BASIC".

danmois commented on projects/adt75/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,91 @@
+		if (ret)
+			goto error;
+
+		printf("Temperature: %.03f C\n", (double)val / 1000);
```
Maybe using pr_info instead of printf would be better.

CiprianRegus commented:
- Removed the `adt75_reg_size()` function, and computed the register size inline.
- Moved the addition of the `MILLIDEGREE_PER_DEGREE` to another commit.
- Replaced the `printf()` calls with `pr_info()` in the example project.
- Fixed the warnings in the IIO driver.
- Removed the channels update function from the IIO driver, since there is only 1 channel (which hardware wise, it's always enabled).

amiclaus commented on drivers/temperature/adt75/adt75.c:
```diff
@@ -0,0 +1,179 @@
+ * @param desc - The device structure.
+ * @param addr - The register's address.
+ * @param val - The value to be read.
+ * @return 0 in case of success, -EINVAL otherwise
```
nit: are you sure the negative return code will always be EINVAL?

amiclaus commented on drivers/temperature/adt75/adt75.c:
```diff
@@ -0,0 +1,179 @@
+
+	reg_size = (addr == ADT75_CONF_REG)? 1 : 2;
+
+	desc->buff[0] = no_os_field_get(ADT75_DATA_REG_MASK, addr);
```
can we reduce the scope of `buff` to the read/write function and drop it from the device structure?

amiclaus commented on drivers/temperature/adt75/adt75.c:
```diff
@@ -0,0 +1,179 @@
+
+	reg_val = no_os_field_get(ADT75_TEMP_MASK, reg_val);
+	*val = no_os_sign_extend32(reg_val, ADT75_SIGN_BIT);
+	*val *= MILLIDEGREE_PER_DEGREE;
```
nit: can reduce here the number of lines: `*val = *val * MILLIDEGREE_PER_DEGREE / ADT75_TEMP_DIV;`

amiclaus commented on drivers/temperature/adt75/adt75.h:
```diff
@@ -0,0 +1,105 @@
+#define ADT75_ONESHOT_MASK		NO_OS_BIT(5)
+
+#define ADT75_HYST_MASK			NO_OS_GENMASK(15, 4)
+#define ADT75_OVER_TEMP_MASK	NO_OS_GENMA.SK(15, 4)
```
typo here `NO_OS_GENMASK(15, 4)`

buha commented on drivers/temperature/adt75/adt75.h:
```diff
@@ -0,0 +1,105 @@
+#ifndef _ADT75_H
+#define _ADT75_H
+
+#include "stdint.h"
```
use <> here

buha commented on projects/adt75/Makefile:
```diff
@@ -0,0 +1,9 @@
+# Select the example you want to enable by chossing y for enabling and n for disabling
+DUMMY_EXAMPLE = y
```
yeah, as ohers have said, maybe rename DUMMY to BASIC

danmois commented on projects/adt75/src/examples/dummy/dummy_example.c:
```diff
@@ -0,0 +1,91 @@
+
+	while (1) {
+		/* Clear the screen. */
+		pr_info("%c",27);
```
Maybe printf should be used here. Problems could arise when using timestamps for pr_info.

rbolboac commented on projects/adt75/src/examples/iio_example/iio_example.c:
```diff
@@ -0,0 +1,92 @@
+/******************************************************************************/
+/************************ Variable Declarations ******************************/
+/******************************************************************************/
+uint8_t iio_data_buffer[DATA_BUFFER_SIZE * sizeof(int32_t)];
```
taking into consideration that the scan type is int16_t, it would make sense to allocate a buffer with the same size multiplier

rbolboac commented on projects/adt75/src/examples/iio_example/iio_example.c:
```diff
@@ -0,0 +1,92 @@
+	struct adt75_iio_init_param adt75_iio_ip;
+	struct iio_data_buffer buff = {
+		.buff = (void *)iio_data_buffer,
+		.size = DATA_BUFFER_SIZE * sizeof(uint32_t)
```
taking into consideration that the scan type is int16_t, it would make sense to obtain the size of the buffer with the same type size multiplier

rbolboac commented on drivers/temperature/adt75/adt75.c:
```diff
@@ -0,0 +1,179 @@
+ *   @brief  Source file of ADT75 Driver.
+ *   @author Ciprian Regus (ciprian.regus@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
```
even though this is an older pull request, it would be nice to change the year to 2023 in all files

amiclaus commented on drivers/temperature/adt75/adt75.h:
```diff
@@ -0,0 +1,104 @@
+#define ADT75_CONF_REG          0x1
+#define ADT75_HYST_REG          0x2
+#define ADT75_OS_REG            0x3
+#define ADT75_ONE_SHOT_REG      0x4s
```
was `s` added on purpose?

CiprianRegus commented:
- Changed size of the IIO buffer, according to the sample size.
- 2022 -> 2023
- removed the `s` character.
